Modular Scripts/File Monitoring/configure-file-monitoring.sh

```diff
@@ -400,49 +403,94 @@ write21ConfFileContents()
 	logMsgToConfigSysLog "INFO" "INFO: Creating file $FILE_SYSLOG_CONFFILE"
 	sudo touch $FILE_SYSLOG_CONFFILE
 	sudo chmod o+w $FILE_SYSLOG_CONFFILE
-
-	imfileStr="
-	\$ModLoad imfile
-	\$InputFilePollInterval 10
-	\$WorkDirectory $RSYSLOG_DIR
-	"
-	if [[ "$LINUX_DIST" == *"Ubuntu"* ]]; then
-		imfileStr+="\$PrivDropToGroup adm
-		"
-	fi
 	
 	rsyslog_version="$(rsyslogd -v)"
 	r_ver=${rsyslog_version:9:1}
-	if [ $r_ver -le 6 ] 
+	if [ $r_ver -le 7 ] 
 	then
-		imfileStr+="
-		# File access file:
-		\$InputFileName $FILE_TO_MONITOR
-		\$InputFileTag $LOGGLY_FILE_TO_MONITOR_ALIAS:
-		\$InputFileStateFile stat-$STATE_FILE_ALIAS
-		\$InputFileSeverity info
-		\$InputFilePersistStateInterval 20000
-		\$InputRunFileMonitor
-		#Add a tag for file events
-		\$template $CONF_FILE_FORMAT_NAME,\"<%pri%>%protocol-version% %timestamp:::date-rfc3339% %HOSTNAME% %app-name% %procid% %msgid% [$LOGGLY_AUTH_TOKEN@41058 $TAG] %msg%\n\"
-		if \$programname == '$LOGGLY_FILE_TO_MONITOR_ALIAS' then @@logs-01.loggly.com:514;$CONF_FILE_FORMAT_NAME
-		if \$programname == '$LOGGLY_FILE_TO_MONITOR_ALIAS' then ~
+		imfileStr="
+			\$ModLoad imfile
+			\$InputFilePollInterval 10
+			\$WorkDirectory $RSYSLOG_DIR
+			\$ActionSendStreamDriver gtls
+			\$ActionSendStreamDriverMode 1
+			\$ActionSendStreamDriverAuthMode x509/name
+			\$ActionSendStreamDriverPermittedPeer *.loggly.com
+
+			#RsyslogGnuTLS
+			\$DefaultNetstreamDriverCAFile /etc/rsyslog.d/keys/ca.d/logs-01.loggly.com_sha12.crt
+
+			# File access file:
+			\$InputFileName $FILE_TO_MONITOR
+			\$InputFileTag $LOGGLY_FILE_TO_MONITOR_ALIAS
+			\$InputFileStateFile stat-$STATE_FILE_ALIAS
+			\$InputFileSeverity info
+			\$InputFilePersistStateInterval 20000
+			\$InputRunFileMonitor
+			
+			#Add a tag for file events
+			template (name=\"$CONF_FILE_FORMAT_NAME\" type=\"string\" string=\"<%pri%>%protocol-version% %timestamp:::date-rfc3339% %HOSTNAME% %app-name% %procid% %msgid% [$LOGGLY_AUTH_TOKEN@41058 $TAG] %msg%\n\")
```

Comment:
How is $TAG set?

Reply:
@mostlyjason, From the very beginning, the default tag i.e. **file** is set into the **LOGGLY_FILE_TAG** variable if the user has not provided one externally using the parameter **-tag**  in the command. Please refer the below code line-

Ref: https://github.com/loggly/install-script/blob/master/Modular%20Scripts/File%20Monitoring/configure-file-monitoring.sh#L37

Now, the user can pass multiple tags externally which get stored into the same **LOGGLY_FILE_TAG** variable separated by a comma and iterate inside the **addTagsInConfiguration** function and finally set into the **$TAG** variable. Please refer the below code line-

Ref: https://github.com/loggly/install-script/blob/master/Modular%20Scripts/File%20Monitoring/configure-file-monitoring.sh#L347

Once **$TAG** is set with either one or multiple tags, the same variable is used in the file-monitoring configuration. Please refer the below code line-

Ref: https://github.com/loggly/install-script/blob/master/Modular%20Scripts/File%20Monitoring/configure-file-monitoring.sh#L427

This is how $TAG set.

Reply:
Okay we should document how to set tags in https://www.loggly.com/docs/configure-syslog-script/. I don't see any descriptions for that parameter

Reply:
@mostlyjason, I have added a **Note** to tell the users that how they can set the **tag** for all three **file-monitoring**, **Nginx** and **Apache** script on the document https://www.loggly.com/docs/configure-syslog-script/. I added this information after the **Script Usage** section, please review.